lib/generators/templates/devise.rb

```diff
@@ -103,15 +103,22 @@
   # config.reload_routes = true
 
   # ==> Configuration for :database_authenticatable
-  # For bcrypt, this is the cost for hashing the password and defaults to 12. If
-  # using other algorithms, it sets how many times you want the password to be hashed.
-  #
-  # Limiting the stretches to just one in testing will increase the performance of
-  # your test suite dramatically. However, it is STRONGLY RECOMMENDED to not use
-  # a value less than 10 in other environments. Note that, for bcrypt (the default
-  # algorithm), the cost increases exponentially with the number of stretches (e.g.
-  # a value of 20 is already extremely slow: approx. 60 seconds for 1 calculation).
-  config.stretches = Rails.env.test? ? 1 : 12
+  # For bcrypt (the default algorithm), this is the cost for hashing the password.
+  # If using other algorithms, it sets how many times you want the password to be hashed.
+  config.stretches = if Rails.env.test?
+    # Limiting the stretches to the minimum cost supported by the algorithm
+    # will increase the performance of your test suite dramatically.
+    #
+    # For bcrypt, the minimum cost is 4. For other algorithms it defaults to 1.
```

Comment:
> For bcrypt, the minimum cost is 4.

This is news to me and probably many others, given the old default of 1 here. Is this documented anywhere? What happens if you use a value less than 4?

Reply:
Thanks for the questions. I thought the comment answers the question, but probably this should be clarified. Can you move your question to https://github.com/heartcombo/devise/pull/5092, so we proceed there? @ghiculescu